astropy/visualization/wcsaxes/core.py

```diff
@@ -451,11 +455,21 @@ def draw(self, renderer, inframe=False):
 
         self._drawn = True
 
-    def set_xlabel(self, label, labelpad=1, **kwargs):
-        self.coords[self._x_index].set_axislabel(label, minpad=labelpad, **kwargs)
+    if MATPLOTLIB_LT_30:
+
+        def set_xlabel(self, label, labelpad=1, **kwargs):
+            self.coords[self._x_index].set_axislabel(label, minpad=labelpad, **kwargs)
+
+        def set_ylabel(self, label, labelpad=1, **kwargs):
+            self.coords[self._y_index].set_axislabel(label, minpad=labelpad, **kwargs)
+
+    else:
+
+        def set_xlabel(self, xlabel, labelpad=1, **kwargs):
```

Comment:
These stanzas confuse me: they seem identical except for the change in variable name. Is the problem that `set_xlabel` etc get called with `xlabel='...'` from other code? If so, definitely add a comment to that extent!

Reply:
I've simplified the code to avoid the branching